portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -50,10 +50,17 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String newComment = request.getParameter("new-comment");
-    if (!newComment.isEmpty()){
+    long timestamp = System.currentTimeMillis();
+
+    if (newComment.isEmpty()){
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+    }
+    else{
       Entity commentEntity = new Entity("Comment");
       commentEntity.setProperty("content", newComment);
+      commentEntity.setProperty("timestamp", timestamp);
       this.datastore.put(commentEntity);
+      response.sendRedirect("/index.html");
     }
```

Comment:
The fewer indented lines of code, the easier it is to read code. Typically, when you have

```
if (check for error #1) {
  express that error #1 occurred
} else if (check for error #2) {
  express that error #2 occurred
} else if (check for error #3) {
  express that error #3 occurred
} else {
  handle correct implementation
}
Return
```

It is is easier to read if for every error case you leave "early" (or seems like it leaves early so the reader doesn't have to keep reading further.

```
if (check for error #1) {
  express that error #1 occurred
  return
} 

if (check for error #2) {
  express that error #2 occurred
  return
}

if (check for error #3) {
  express that error #3 occurred
} 

handle correct implementation
Return
```

See a better explanation in the tott I sent in the chat.